From 4325946fcb3b90e93cc7e06f2981999961d1acee Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Mon, 26 Nov 2018 12:41:31 +0100
Subject: [PATCH] PONRTSYS-2772: Add pondef as parent of g_pon

The PON gate clock is after the FOUT2 of the LPLL3, reflect this in the driver.

The gate clock should be switched off before the FOUT2 is switched off because
the gate clocks can switch glitch free, the FOUT cannot switch glitch free.

Fixes: a5bef0a4794 ("DRVLIB_SW-938: CGU-Power-saving-feature")
---
 drivers/clk/intel/clk-prx300.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/intel/clk-prx300.c b/drivers/clk/intel/clk-prx300.c
index c74d8bd2e16c..189bd0012e09 100644
--- a/drivers/clk/intel/clk-prx300.c
+++ b/drivers/clk/intel/clk-prx300.c
@@ -241,7 +241,7 @@ static const struct intel_clk_branch prx300_branch_clks[] __initconst = {
 		   G_DMA3_SHIFT, GATE_CLK_HW, 0),
 	INTEL_GATE(PRX300_GCLK_SWITCH, "g_switch", NULL, 0, CGU_GATE1,
 		   G_SWITCH_SHIFT, GATE_CLK_HW, 0),
-	INTEL_GATE(PRX300_GCLK_PON, "g_pon", NULL, 0, CGU_GATE1,
+	INTEL_GATE(PRX300_GCLK_PON, "g_pon", "pondef", 0, CGU_GATE1,
 		   G_PON_SHIFT, GATE_CLK_HW, 0),
 	INTEL_GATE(PRX300_GCLK_AON, "g_aon", NULL, 0, CGU_GATE1,
 		   G_AON_SHIFT, GATE_CLK_HW, 0),
